miden-tx/tests/test_miden_faucet_contract.rs

```diff
@@ -0,0 +1,433 @@
+
+    let expected_note = NoteStub::new(
+        recipient.into(),
+        NoteVault::new(&[fungible_asset_1]).unwrap(),
+        NoteMetadata::new(faucet_account.id(), Felt::new(4), Felt::new(1)),
+    )
+    .unwrap();
+
+    assert!(transaction_result.created_notes().notes()[0].recipient() == expected_note.recipient());
+    assert!(transaction_result.created_notes().notes()[0].vault() == expected_note.vault());
+    assert!(transaction_result.created_notes().notes()[0].metadata() == expected_note.metadata());
+}
+
+#[test]
+fn test_faucet_contract_mint_fungible_asset_fails_exceeds_max_supply() {
+    let faucet_account_id = AccountId::try_from(ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN).unwrap();
+    let faucet_account_code_src = include_str!("../../miden-lib/asm/faucets/basic.masm");
+    let faucet_account_code_ast = ModuleAst::parse(faucet_account_code_src).unwrap();
+    let mut account_assembler = assembler();
+
+    let faucet_account_code = AccountCode::new(
+        faucet_account_id,
+        faucet_account_code_ast.clone(),
+        &mut account_assembler,
+    )
+    .unwrap();
+
+    let mut faucet_account_storage = mock_account_storage();
+    let faucet_storage_slot_1 = [Felt::new(200), Felt::new(0), Felt::new(0), Felt::new(0)];
+    faucet_account_storage.set_item(1, faucet_storage_slot_1);
+
+    let faucet_account: Account = Account::new(
+        faucet_account_id,
+        AccountVault::new(&vec![]).unwrap(),
+        faucet_account_storage.clone(),
+        faucet_account_code.clone(),
+        Felt::new(1),
+    );
+
+    let (account, block_header, chain, notes) = mock_inputs_with_existing(
+        MockAccountType::FungibleFaucet(ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN),
+        AssetPreservationStatus::Preserved,
+        Some(faucet_account),
+        None,
+    );
+
+    let recipient = [Felt::new(0), Felt::new(1), Felt::new(2), Felt::new(3)];
+    let tag = Felt::new(4);
+    let amount = Felt::new(250);
+
+    let transaction_code = format!(
+        "
+        use.miden::faucets::basic->faucet
+        use.miden::sat::internal::prologue
+
+        begin
+            # prepare the transaction
+            exec.prologue::prepare_transaction
+            
+            push.{recipient}
+            push.{tag}
+            push.{amount}
+            call.faucet::distribute
+            
+        end
+        ",
+        recipient = prepare_word(&recipient),
+        tag = tag,
+        amount = amount,
+    );
+
+    let transaction = prepare_transaction(
+        account,
+        None,
+        block_header,
+        chain,
+        notes,
+        &transaction_code,
+        "",
+        None,
+        None,
+    );
+
+    let mut advice_provider = MemAdviceProvider::from(transaction.advice_provider_inputs());
+
+    let process = run_tx(
+        transaction.tx_program().clone(),
+        transaction.stack_inputs(),
+        &mut advice_provider,
+    );
```

Comment:
Is there a reason we do this instead of `TransactionExecutor::execute_transaction()` as we did in the previous tests?